internal/node/machine_mgr_events.go

```diff
@@ -150,31 +149,57 @@ func (m *MachineManager) publishMachineStopped(vm *runningFirecracker) error {
 	return nil
 }
 
-// Called when the node server gets a log entry via internal NATS. Used to
-// package and re-mit with additional metadata on $NEX.logs...
-func (m *MachineManager) handleAgentLog(msg *nats.Msg) {
-	tokens := strings.Split(msg.Subject, ".")
-	vmID := tokens[1]
-
-	vm, ok := m.allVMs[vmID]
+// // Called when the node server gets a log entry via internal NATS. Used to
+// // package and re-mit with additional metadata on $NEX.logs...
+// func (m *MachineManager) handleAgentLog(msg *nats.Msg) {
+// 	tokens := strings.Split(msg.Subject, ".")
+// 	vmID := tokens[1]
+
+// 	vm, ok := m.allVMs[vmID]
+// 	if !ok {
+// 		m.log.Warn("Received a log message from an unknown VM.")
+// 		return
+// 	}
+
+// 	var logentry agentapi.LogEntry
+// 	err := json.Unmarshal(msg.Data, &logentry)
+// 	if err != nil {
+// 		m.log.Error("Failed to unmarshal log entry from agent", slog.Any("err", err))
+// 		return
+// 	}
+
+// 	m.log.Debug("Received agent log", slog.String("vmid", vmID), slog.String("log", logentry.Text))
+
+// 	bytes, err := json.Marshal(&emittedLog{
+// 		Text:      logentry.Text,
+// 		Level:     slog.Level(logentry.Level),
+// 		MachineId: vmID,
+// 	})
+// 	if err != nil {
+// 		m.log.Error("Failed to marshal our own log entry", slog.Any("err", err))
+// 		return
+// 	}
+
+// 	var workload *string
+// 	if vm.deployRequest != nil {
+// 		workload = vm.deployRequest.WorkloadName
+// 	}
+
+// 	subject := logPublishSubject(vm.namespace, m.node.publicKey, vmID, workload)
+// 	_ = m.nc.Publish(subject, bytes)
+// }
+
+func (m *MachineManager) agentLog(agentId string, entry agentapi.LogEntry) {
+	vm, ok := m.allVMs[agentId]
```

Comment:
The map of VMs should still be indexed on vm id. Did we just wholesale rename vm id to agent id?

Instead, we should probably index these based on workload id. Which should come from the `WorkloadManager` eventually...